bowtie/_cli.py

```diff
@@ -727,12 +727,66 @@ def validate(
 
 @implementation_subcommand()  # type: ignore[reportArgumentType]
 @FORMAT
-async def info(implementations: Iterable[Implementation], format: _F):
+@click.option(
+    "--no-remote-data",
+    is_flag=True,
+    default=False,
+    help="Disable retrieving implementation (harness)'s metadata from GitHub",
+)
+async def info(
+    implementations: Iterable[Implementation],
+    format: _F,
+    no_remote_data: bool,
+):
     """
     Retrieve a particular implementation (harness)'s metadata.
     """
     for each in implementations:
         metadata = [(k, v) for k, v in each.info.serializable().items() if v]
+        if not no_remote_data:
+            try:
+                import re
+
+                from github3 import GitHub  # type: ignore[reportMissingTypeStubs]
+                from github3.exceptions import (  # type: ignore[reportMissingTypeStubs]
+                    ForbiddenError,
+                    NotFoundError,
+                )
+
+                gh = GitHub(token=os.environ.get("GITHUB_TOKEN", ""))
+                pattern = r"https://github\.com/([^/]+)/([^/]+)"
+                metadata_dict = dict(metadata)
+                if "source" in metadata_dict:
+                    match = re.search(pattern, metadata_dict.get("source"))  # type: ignore[reportUnknownMemberType,reportUnknownArgumentType]
+                    if match:
+                        org = match.group(1)  # type: ignore[reportUnknownMemberType,reportUnknownVariableType]
```

Comment:
This looks too long to do inline here in this function where it's taking up a significant portion of space even though it doesn't really have to do with the core of what this command does. Try to think about splitting it into a helper function.

Reply:
I think this will get resolved once I figure out how to make no_remote_data as a callable.